src/main/java/org/apache/commons/lang3/Functions.java

```diff
@@ -536,23 +545,22 @@ public static void tryWithResources(FailableRunnable<? extends Throwable> pActio
     }
 
     /**
-     * Rethrow a {@link Throwable} as an unchecked exception.
+     * Rethrows a {@link Throwable} as an unchecked exception.
      * @param pThrowable The throwable to rethrow
      * @return Never returns anything, this method never terminates normally
      */
     public static RuntimeException rethrow(Throwable pThrowable) {
         if (pThrowable == null) {
             throw new NullPointerException("The Throwable must not be null.");
+        }
+        if (pThrowable instanceof RuntimeException) {
```

Comment:
This is a completely different change; please don't mix changes in a pull request.

Reply:
Will commit to master instead. A PR for this is over the top IMO.